File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int addDigits(int num) {
        int sum = 0;
        while(num > 0) {
            int digit = num % 10;
            sum =  sum + digit;
            num = num / 10;
            if(num == 0) {
                if( sum < 10 ) { return sum; }
                num = sum;
                sum = 0;
            }
        }
        return 0;
    }
}

File: 0258-add-digits/0258-add-digits.go
func addDigits(num int) int {
    sum := 0
    for num > 0 {
        digit := num % 10
        sum += digit
        num /= 10
        if num == 0 {
            if sum < 10 { return sum; }
            num = sum
            sum = 0
        }
    }
    return 0
}